addons/pos_self_order/models/product_product.py

```diff
@@ -52,12 +52,31 @@ def _load_pos_self_data_fields(self, config_id):
 
     def _load_pos_self_data(self, data):
         domain = self._load_pos_self_data_domain(data)
-        fields = self._load_pos_self_data_fields(data['pos.config']['data'][0]['id'])
+        config_id = data['pos.config']['data'][0]['id']
+        fields = self._load_pos_self_data_fields(config_id)
```

Comment:
you do not have to change those two lines

Reply:
or you can keep it and only pass the `config_id` to the `_compute_product_price_with_pricelist` which could be even better what do you think ?

Reply:
@adgu-odoo Thanks for reviewing my code. It was my initial plan. Now, I've modified the code.